Sources/Rendering/Core/Picker/index.js

```diff
@@ -89,9 +89,171 @@ function vtkPicker(publicAPI, model) {
+   * @param {*} p2World
+   * @returns true if we picked something else false
+   */
+  function pick3DInternal(renderer, tolerance, p1World, p2World) {
+    const p1Mapper = new Float64Array(4);
+    const p2Mapper = new Float64Array(4);
+    const t = [];
+
+    const ray = [];
+    let pickable;
+    const hitPosition = [];
+
+    if (!renderer) {
+      vtkErrorMacro('You must specify a renderer.');
+      return;
+    }
+
+    //  Loop over all props.  Transform ray (defined from position of
+    //  camera to selection point) into coordinates of mapper (not
+    //  transformed to actors coordinates!  Reduces overall computation!!!).
+    //  Note that only vtkProp3D's can be picked by vtkPicker.
+    const props = model.pickFromList ? model.pickList : renderer.getActors();
+
+    const scale = [];
+    props.forEach((prop) => {
+      const mapper = prop.getMapper();
+      pickable =
+        prop.getNestedPickable() &&
+        prop.getNestedVisibility() &&
+        prop.getProperty().getOpacity() > 0.0;
+
+      //  If actor can be picked, get its composite matrix, invert it, and
+      //  use the inverted matrix to transform the ray points into mapper
+      //  coordinates.
+      if (pickable) {
+        model.transformMatrix = prop.getMatrix().slice(0);
+
+        mat4.transpose(model.transformMatrix, model.transformMatrix);
+        mat4.invert(model.transformMatrix, model.transformMatrix);
+
+        // Extract scale
+        const col1 = [
+          model.transformMatrix[0],
+          model.transformMatrix[1],
+          model.transformMatrix[2],
+        ];
+        const col2 = [
+          model.transformMatrix[4],
+          model.transformMatrix[5],
+          model.transformMatrix[6],
+        ];
+        const col3 = [
+          model.transformMatrix[8],
+          model.transformMatrix[9],
+          model.transformMatrix[10],
+        ];
+
+        scale[0] = vtkMath.norm(col1);
+        scale[1] = vtkMath.norm(col2);
+        scale[2] = vtkMath.norm(col3);
+
+        vec4.transformMat4(p1Mapper, p1World, model.transformMatrix);
+        vec4.transformMat4(p2Mapper, p2World, model.transformMatrix);
+
+        p1Mapper[0] /= p1Mapper[3];
+        p1Mapper[1] /= p1Mapper[3];
+        p1Mapper[2] /= p1Mapper[3];
+
+        p2Mapper[0] /= p2Mapper[3];
+        p2Mapper[1] /= p2Mapper[3];
+        p2Mapper[2] /= p2Mapper[3];
+
+        for (let i = 0; i < 3; i++) {
+          ray[i] = p2Mapper[i] - p1Mapper[i];
+        }
+
+        //  Have the ray endpoints in mapper space, now need to compare this
+        //  with the mapper bounds to see whether intersection is possible.
+        //
+        //  Get the bounding box of the modeller.  Note that the tolerance is
+        //  added to the bounding box to make sure things on the edge of the
+        //  bounding box are picked correctly.
+        const bounds = mapper
+          ? vtkBoundingBox.inflate(mapper.getBounds(), tolerance)
+          : [0, 0, 0, 0, 0, 0];
```

Comment:
`[0,0,0,0,0,0]` is a valid bounding box. There is a constant (or convenient function ) that creates uninvalid/uninitialized bounds

Reply:
> `[0,0,0,0,0,0]` is a valid bounding box. There is a constant (or convenient function ) that creates uninvalid/uninitialized bounds

The symbol is [INIT_BOUNDS](https://github.com/Kitware/vtk-js/blob/master/Sources/Common/DataModel/BoundingBox/index.js#L5-L12)